scripts/styleChecker.fsx

```diff
@@ -0,0 +1,442 @@
+                Arguments = "diff --exit-code"
+            },
+            Echo.Off
+        )
+
+    processResult
+
+let GitRestore() =
+    Process
+        .Execute(
+            {
+                Command = "git"
+                Arguments = "restore ."
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let CheckStyleOfFSharpFiles(rootDir: DirectoryInfo) : bool =
+    let suggestion =
+        Some "Please style your F# code using: `dotnet fantomless --recurse .`"
+
+    GitRestore()
+
+    let success =
+        if ContainsFiles rootDir "*.fs" || ContainsFiles rootDir ".fsx" then
+            StyleFSharpFiles rootDir
+            let processResult = GitDiff()
+            UnwrapProcessResult suggestion true processResult |> ignore
+            IsProcessSuccessful processResult
+
+        else
+            true
+
+    success
+
+let CheckStyleOfTypeScriptFiles(rootDir: DirectoryInfo) : bool =
+    let pattern = Path.Combine [| "."; "**"; "*.ts" |]
+
+    let suggestion =
+        Some
+            $"Please style your TypeScript code using: `npx prettier --quote-props=consistent --write {pattern}`"
+
+    GitRestore()
+
+    let success =
+        if ContainsFiles rootDir "*.ts" then
+            InstallPrettier prettierVersion
+            StyleTypeScriptFiles()
+            let processResult = GitDiff()
+            UnwrapProcessResult suggestion true processResult |> ignore
+            IsProcessSuccessful processResult
+
+        else
+            true
+
+    success
+
+let CheckStyleOfYmlFiles(rootDir: DirectoryInfo) : bool =
+    let pattern = Path.Combine [| "."; "**"; "*.yml" |]
+
+    let suggestion =
+        Some
+            $"Please style your YML code using: `npx prettier --quote-props=consistent --write {pattern}`"
+
+    GitRestore()
+
+    let success =
+        if ContainsFiles rootDir "*.yml" then
+            InstallPrettier prettierVersion
+            StyleYmlFiles()
+            let processResult = GitDiff()
+            UnwrapProcessResult suggestion true processResult |> ignore
+            IsProcessSuccessful processResult
+        else
+            true
+
+    success
+
+let CheckStyleOfCSharpFiles(rootDir: DirectoryInfo) : bool =
+    let suggestion =
+        Some
+            "Please style your C# code using: `dotnet format whitespace . --folder"
+
+    GitRestore()
+
+    let success =
+        if ContainsFiles rootDir "*.cs" then
+            StyleCSharpFiles rootDir
+            let processResult = GitDiff()
+            UnwrapProcessResult suggestion true processResult |> ignore
+            IsProcessSuccessful processResult
+        else
+            true
+
+    success
+
+let CheckStyleOfXamlFiles(rootDir: DirectoryInfo) : bool =
+    let prettierPath =
+        Path.Combine
+            [|
+                "."
+                "node_modules"
+                ".bin"
+                "prettier"
+            |]
+
+    let pattern = Path.Combine [| "**"; "*.xaml" |]
```

Comment:
@realmarv a pattern is not a path, please don't use Path.Combine here, use Path.DirectorySeparatorChar

Reply:
Done